akka-http-tests/src/test/scala/akka/http/javadsl/server/HttpAppSpec.scala

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.http.javadsl.server
+
+import java.util.concurrent.TimeUnit
+
+import akka.http.javadsl.settings.ServerSettings
+import akka.http.scaladsl.{ Http, TestUtils }
+import akka.http.scaladsl.client.RequestBuilding
+import akka.http.scaladsl.model.{ HttpRequest, StatusCodes }
+import akka.stream.ActorMaterializer
+import akka.testkit.AkkaSpec
+import com.typesafe.config.ConfigFactory
+import org.scalatest.concurrent.Eventually
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{ Await, Future }
+import scala.runtime.BoxedUnit
+
+class HttpAppSpec extends AkkaSpec with RequestBuilding with Eventually {
+  import system.dispatcher
+
+  "HttpApp" should {
+
+    "start without ActorSystem" in {
+      val (_, host, port) = TestUtils.temporaryServerHostnameAndPort()
+
+      val minimal = new MinimalHttpApp()
+
+      val server = Future {
+        minimal.startServer(host, port, ServerSettings.create(ConfigFactory.load))
+      }
+
+      verifyRequest(host, port)
+
+      minimal.shutdownTrigger.complete(BoxedUnit.UNIT)
```

Comment:
>show in the app that you trigger this somehow, add a route that does /shutdown and triggers this.
>Otherwise some people will certainly be confused ;)
Here is how we trigger the shutdown. In this case, we decide to shutdown the server from the "outside". I can add some comments in here. Is that what you mean?